theano/tensor/blas.py

```diff
@@ -182,6 +182,68 @@
                         'dot(vector, vector) (%s)',
                         str(e))
 
+def check_force_gemv_init_core(opt):
```

Comment:
I think we should change opt to pass the op we want to test and pass opt None, to make sure we test the right op.

This would also fix the current recursion.